src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTag.cs

```diff
@@ -269,13 +273,14 @@ private void Adornment_MouseLeftButtonDown(object sender, MouseButtonEventArgs e
                 var replacementValue = _hint.ReplacementTextChange!.Value;
                 var subjectBuffer = _span.Snapshot.TextBuffer;
 
-                if (subjectBuffer.CurrentSnapshot.Length > replacementValue.Span.End)
+                var updatedSpan = _trackingSpan!.GetSpan(subjectBuffer.CurrentSnapshot);
```

Comment:
i don't think youneed this.  you have a _span to begin with.  it can be tracked forward to the current ITextSnapshot right here.  Happy to discuss how this works tomorrow if you're interested :)